### PR TITLE
pass CLAIM_URI_LOCALE value from identity.xml

### PR DESCRIPTION
**Pull request by ravindudsilva**
### Proposed changes in this pull request

Update the value of `CLAIM_URI_LOCALE` to  `http://wso2.org/claims/local`, valued passed from identity.xml

Related issue - https://github.com/wso2/product-is/issues/12163

### CODE REVIEW

File: components/event-handler-notification/org.wso2.carbon.identity.event.handler.notification/src/main/java/org/wso2/carbon/identity/event/handler/notification/NotificationConstants.java [hunk 1]
```diff
@@ -18,13 +18,15 @@
...
+import org.wso2.carbon.identity.core.util.IdentityUtil;
+
 public class NotificationConstants {
 
     private NotificationConstants() {
     }
 
     public static final String TEMPLATE_TYPE = "TEMPLATE_TYPE";
-    public static final String CLAIM_URI_LOCALE = "http://wso2.org/claims/locality";
+    public static final String CLAIM_URI_LOCALE = IdentityUtil.getProperty("LocaleClaimUri");
```
Comment (emswbandara) [inline components/event-handler-notification/org.wso2.carbon.identity.event.handler.notification/src/main/java/org/wso2/carbon/identity/event/handler/notification/NotificationConstants.java:29]: shall we move the claim fetch part to a private method? 
Assign a default value if config is null.

Comment (ravindudsilva) [reply components/event-handler-notification/org.wso2.carbon.identity.event.handler.notification/src/main/java/org/wso2/carbon/identity/event/handler/notification/NotificationConstants.java]: Fixed in commit 571de68b7b4b2b6c45f7a6d35b7b61bd5512c9b4

File: components/event-handler-notification/org.wso2.carbon.identity.event.handler.notification/src/main/java/org/wso2/carbon/identity/event/handler/notification/NotificationConstants.java [hunk 2]
```diff
@@ -18,13 +18,23 @@
...
 package org.wso2.carbon.identity.event.handler.notification;
 
+import org.wso2.carbon.identity.core.util.IdentityUtil;
+
 public class NotificationConstants {
 
     private NotificationConstants() {
     }
 
+    private static String getClaimUriLocale() {
```
Comment (emswbandara) [inline components/event-handler-notification/org.wso2.carbon.identity.event.handler.notification/src/main/java/org/wso2/carbon/identity/event/handler/notification/NotificationConstants.java:28]: add new line after method signature

File: components/event-handler-notification/org.wso2.carbon.identity.event.handler.notification/src/main/java/org/wso2/carbon/identity/event/handler/notification/NotificationConstants.java [hunk 3]
```diff
@@ -18,13 +18,23 @@
...
+        if ("true".equals(IdentityUtil.getProperty("isLegacyLocalityClaimUri"))) {
```
Comment (emswbandara) [inline components/event-handler-notification/org.wso2.carbon.identity.event.handler.notification/src/main/java/org/wso2/carbon/identity/event/handler/notification/NotificationConstants.java:29]: better to use Boolean.parseBoolean(IdentityUtil.getProperty("isLegacyLocalityClaimUri"))

Comment (emswbandara) [reply components/event-handler-notification/org.wso2.carbon.identity.event.handler.notification/src/main/java/org/wso2/carbon/identity/event/handler/notification/NotificationConstants.java:29]: also first check whether IdentityUtil.getProperty("isLegacyLocalityClaimUri") is null, if so, assign the default value.

Comment (ravindudsilva) [reply components/event-handler-notification/org.wso2.carbon.identity.event.handler.notification/src/main/java/org/wso2/carbon/identity/event/handler/notification/NotificationConstants.java]: commit c6f1d06820885092027e0f28adc4ac8b890cce3e

File: components/event-handler-notification/org.wso2.carbon.identity.event.handler.notification/src/main/java/org/wso2/carbon/identity/event/handler/notification/NotificationConstants.java (hunk 2 above)
Comment (ravindudsilva) [reply components/event-handler-notification/org.wso2.carbon.identity.event.handler.notification/src/main/java/org/wso2/carbon/identity/event/handler/notification/NotificationConstants.java:28]: Fixed in commit c6f1d06820885092027e0f28adc4ac8b890cce3e

File: components/event-handler-notification/org.wso2.carbon.identity.event.handler.notification/src/main/java/org/wso2/carbon/identity/event/handler/notification/NotificationConstants.java [hunk 4]
```diff
@@ -18,13 +18,25 @@
...
+
 public class NotificationConstants {
 
     private NotificationConstants() {
     }
 
+    private static String getClaimUriLocale() {
+
+        if (Boolean.parseBoolean(IdentityUtil.getProperty("useLegacyLocalizationClaim"))
+        || IdentityUtil.getProperty("useLegacyLocalizationClaim") == null) {
```
Comment (emswbandara) [inline components/event-handler-notification/org.wso2.carbon.identity.event.handler.notification/src/main/java/org/wso2/carbon/identity/event/handler/notification/NotificationConstants.java:31]: if IdentityUtil.getProperty("useLegacyLocalizationClaim") == null we need to return "http://wso2.org/claims/local"

Comment (IsuraD) [reply components/event-handler-notification/org.wso2.carbon.identity.event.handler.notification/src/main/java/org/wso2/carbon/identity/event/handler/notification/NotificationConstants.java:31]: Yes. if it is not defined, we have to use the new behavior.

Comment (IsuraD) [reply components/event-handler-notification/org.wso2.carbon.identity.event.handler.notification/src/main/java/org/wso2/carbon/identity/event/handler/notification/NotificationConstants.java:31]: We need to check the behavior in challenge questions too. Please check that too.

**Review by emswbandara — APPROVED** [review]

**Pull request merged**